### PR TITLE
20 ai structure

### PR DESCRIPTION
**Pull request by adsamu**

### CODE REVIEW

**Comment by Alexnder77** [comment]
ok!

File: internal/ai/pre-calculator-ugglan.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+type PreCalculator struct {
+}
+
+func NewPreCalculator() *PreCalculator {
+	pc := &PreCalculator{}
+	return pc
+}
+
+type Data struct { // name is a placeholder
```
Comment (Ferazz) [inline internal/ai/pre-calculator-ugglan.go:17]: fix placeholder >:(

File: internal/ai/pre-calculator-ugglan.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+func (pc *PreCalculator) Process(gamestate *gamestate.GameState) *Data {
+	fmt.Println("Ugglan")
```
Comment (Ferazz) [inline internal/ai/pre-calculator-ugglan.go:21]: hoot

**Review by Ferazz — APPROVED** [review]
lgtm

**Pull request merged**